include/net/buf.h

```diff
@@ -934,6 +930,9 @@ struct net_buf {
 	/** Where the buffer should go when freed up. */
 	uint8_t pool_id;
 
+	/* Size of user data on this buffer */
+	uint8_t user_data_len;
```

Comment:
This should be called `user_data_size`. `len` is used (in the net_buf API)  to indicate how much valid data there is in a buffer, while `size` is used to denote the total size of the buffer.

Reply:
done